pkg/util/util.go

```diff
@@ -68,3 +71,23 @@ func CreatePersistanceStorage(sPath, metaDataStore, driverName string) (CachePer
 func createPersistentStorage(persistentStoragePath string) error {
 	return os.MkdirAll(persistentStoragePath, os.FileMode(0755))
 }
+
+// ValidateDriverName validates the driver name
+func ValidateDriverName(driverName string) error {
```

Comment:
Validation comments:
  - We should use the validation API IsDNS1035Label (which includes the length check as well)
  - We should validate trailer as ".csi.ceph.com" so that we retain the domain for any instance

Reply:
current code is taken from the kubernetes core  validation

https://github.com/kubernetes/kubernetes/blob/c094f873a9d1ab444c0834fa3d3adf5ed2f44105/pkg/apis/core/validation/validation.go#L1442-L1458

```
func ValidateCSIDriverName(driverName string, fldPath *field.Path) field.ErrorList {
	allErrs := field.ErrorList{}


	if len(driverName) == 0 {
		allErrs = append(allErrs, field.Required(fldPath, ""))
	}


	if len(driverName) > 63 {
		allErrs = append(allErrs, field.TooLong(fldPath, driverName, 63))
	}


	for _, msg := range validation.IsDNS1123Subdomain(strings.ToLower(driverName)) {
		allErrs = append(allErrs, field.Invalid(fldPath, driverName, msg))
	}


	return allErrs
}
``` 

drivername could be anything, example `org.test.csi-hostpath` etc i dont  think we should validate the trainer as  .csi.ceph.com. if its required i can  do that one

Reply:
It looks like the kubernetes implementation and spec differ? I would stick to the spec for now.